pyproject.toml

```diff
@@ -54,7 +55,7 @@ dev = [
     'setuptools<49.2',
     'bump-my-version>=0.18.3,<1',
     'pip>=9.0.1',
-    'watchdog>=0.8.3,<0.11',
+    'watchdog>=1.0.1,<5',
```

Comment:
Python 3.12 has [removed support ](https://github.com/neovim/pynvim/issues/538#issue-1933484611)for the `imp` library, which `watchdog` uses up to [version 0.10.7.](https://github.com/gorakhargosh/watchdog/commit/86b983aa5f7bf2a65314b5dfb7ff8c1b930c10c4#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7L38)